etna/packages/etna-js/upload/workers/uploader.jsx

```diff
@@ -174,14 +176,18 @@ export default (uploader) => {
 
     remove: (upload) => uploader.dispatch({ type: 'REMOVE_UPLOAD', upload }),
 
-    timeout: () => {
-      ++uploader.timeouts;
-      if (uploader.timeouts == uploader.maxTimeouts) {
-        uploader.reset();
+    timeout: (upload) => {
+      const { url } = upload;
+      const { timeouts, maxTimeouts } = uploader;
+
+      timeouts[url] = timeouts[url] || 0;
+
+      if (++timeouts[url] >= maxTimeouts) {
         uploader.dispatch({ type: 'UPLOAD_TIMEOUT', upload });
```

Comment:
Does anything catch this action and display a message or log it? I don't see anything, but it might be nice to display something.

Reply:
Yeah, so... it doesn't look like it does anything right now lol.  But I am keen on it pausing the download and notifying the user, which I think I can add.  My plan is to do some deeper refactoring that will make this easier.